code/datums/storage/subtypes/rped.dm

```diff
@@ -9,7 +13,81 @@
 	max_specific_storage = WEIGHT_CLASS_NORMAL
 	numerical_stacking = TRUE
 
+	/**
+	 * as of now only these stack components are required to build machines like[thermomaachine,crystallizer,electrolyzer]
+	 * so we limit the rped to pick up only these stack types so players dont cheat and use this as a general storage medium
+	 */
+	var/static/list/allowed_material_types=list(
+		/obj/item/stack/sheet/glass,
+		/obj/item/stack/sheet/plasteel,
+		/obj/item/stack/cable_coil,
+	)
+
+	/**
+	 * we check if the user is trying to insert any of these bluespace crystal types into the RPED
+	 * at any point the total sum of all these types in the RPED must be 30
+	 * for example 10 refined crystals + 15 artifical crystals + 5 sheets=30 or any other combination like this
+	 */
+	var/static/list/allowed_bluespace_types=list(
+		/obj/item/stack/ore/bluespace_crystal,
+		/obj/item/stack/ore/bluespace_crystal/refined,
+		/obj/item/stack/ore/bluespace_crystal/artificial,
+		/obj/item/stack/sheet/bluespace_crystal,
+	)
+
 /datum/storage/rped/can_insert(obj/item/to_insert, mob/user, messages = TRUE, force = FALSE)
 	. = ..()
-	if(!to_insert.get_part_rating())
+
+	//we check how much of glass,plasteel & cable the user can insert
+	if(isstack(to_insert))
+		//user tried to insert invalid stacktype
+		if(!is_type_in_list(to_insert,allowed_material_types) && !is_type_in_list(to_insert,allowed_bluespace_types))
+			return FALSE
+
+		var/obj/item/stack/the_stack = to_insert
+		//how much of the stack is the user trying to insert
```

Comment:
You don't need to comment everything, only what's not obvious

Reply:
Yup i love details my bad ill remove that